queue-jdbc/src/main/java/io/kestra/queue/jdbc/JdbcQueueFactory.java

```diff
@@ -0,0 +1,54 @@
+package io.kestra.queue.jdbc;
+
+import io.kestra.queue.*;
+import io.kestra.queue.jdbc.client.JdbcQueueClient;
+import io.micronaut.context.ApplicationContext;
+import io.micronaut.context.annotation.Context;
+import io.micronaut.inject.qualifiers.Qualifiers;
+import jakarta.annotation.PostConstruct;
+import jakarta.inject.Inject;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Context
+@JdbcQueueEnabled
+public class JdbcQueueFactory {
+    @Inject
+    private QueueUtils queueUtils;
+
+    @Inject
+    private ApplicationContext applicationContext;
+
+    @Inject
+    private JdbcQueueClient jdbcQueueClient;
+
+    @PostConstruct
+    void init() {
```

Comment:
Is it not preferable to use the `@EachBean`Micronaut annotation which is designed for such use case ?  Also, when does Micronaut will close the queue (i.e., call the PreDestroy method) ?